devito/mpi/routines.py

```diff
@@ -513,6 +520,14 @@ class OverlapHaloExchangeBuilder(DiagHaloExchangeBuilder):
         remainder()
     """
 
+    @property
+    def headers(self):
```

Comment:
Here's a potentially better way of achieving the same thing, which generalizes to future Macros as well..

We could have a final IET pass that parses the AST looking for special symbolic objects, such as Min, Max, ... "special" here means for example inheriting from a mixin class "MacroGenerator" or something along the likes of it

Such IET pass would then very very simply return 

```
macro_generators = ...
headers = [n.header for n in macro_generators]

return iet, {'headers': headers}
```

This may or may not be an addition we want to do with this PR, but if not, I would open an issue

By doing that, we wouldn't require each individual pass using a macro having to care about it !

Reply:
That's a good point, not sure the scope of this PR. Not sure which one is easier to maintain/extend some headers/macro may be tricky to infer.

Reply:
The thing I suggested above would be, by experience, easier.

Basically: the more we automate, the better

I agree we can park it for another day though. Could u open an enhancement-issue once we land this PR?